checkov/terraform/graph_builder/local_graph.py

```diff
@@ -172,90 +177,115 @@ def _build_edges(self) -> None:
+                and isinstance(vertex.attributes.get('source')[0], str):
+            target_path = vertex.path
+            if vertex.module_dependency != "":
+                target_path = unify_dependency_path([vertex.module_dependency, vertex.path])
+            dest_module_path = self._get_dest_module_path(
+                curr_module_dir=self.get_dirname(vertex.path),
+                dest_module_source=vertex.attributes["source"][0],
+                dest_module_version=vertex.attributes.get("version", ["latest"])[0]
+            )
+            target_variables = [
+                index
+                for index in self.vertices_by_module_dependency.get(
+                    (target_path, self.module.module_address_map.get((vertex.path, vertex.name))), {}).get(
+                    BlockType.VARIABLE, [])
+                if self.get_dirname(self.vertices[index].path) == dest_module_path
+            ]
+            for attribute in vertex.attributes.keys():
+                if attribute in MODULE_RESERVED_ATTRIBUTES:
+                    continue
+                target_variable = next((v for v in target_variables if self.vertices[v].name == attribute), None)
+                if target_variable is not None:
+                    self._create_edge(target_variable, origin_node_index, "default", cross_variable_edges)
+        elif vertex.block_type == BlockType.TF_VARIABLE:
+            # Assuming the tfvars file is in the same directory as the variables file (best practice)
+            target_variables = [
+                index
+                for index in self.vertices_block_name_map.get(BlockType.VARIABLE, {}).get(vertex.name, [])
+                if self.get_dirname(self.vertices[index].path) == self.get_dirname(vertex.path)
+            ]
+            if len(target_variables) == 1:
+                self._create_edge(target_variables[0], origin_node_index, "default", cross_variable_edges)
+
+    def _build_cross_variable_edges(self):
+        target_nodes_indexes = [v for v, referenced_vertices in self.out_edges.items() if
+                                self.vertices[v].block_type == BlockType.RESOURCE and any(
+            self.vertices[e.dest].block_type != BlockType.RESOURCE for e in referenced_vertices)]
+        aliases = self._get_aliases()
+        resources_types = self.get_resources_types_in_graph()
+        for origin_node_index in target_nodes_indexes:
+            vertex = self.vertices[origin_node_index]
+            self._build_edges_for_vertex(origin_node_index, vertex, aliases, resources_types, True)
+
+    def _create_edge(self, origin_vertex_index: int, dest_vertex_index: int, label: str,
+                     cross_variable_edges: bool = False) -> bool:
         if origin_vertex_index == dest_vertex_index:
-            return
+            return False
         edge = Edge(origin_vertex_index, dest_vertex_index, label)
+        if cross_variable_edges:
+            if any(str(e) == str(edge) for e in self.edges):
```

Comment:
```suggestion
            if any(e == edge for e in self.edges):
```
the `__eq__` in class `Edge` does it already, so no need to do it here 🙂 
```suggestion
            if edge in self.edges:
```
I think this should even work